src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -507,6 +507,24 @@ public override async ValueTask<FormulaValue> Visit(AggregateCoercionNode node,
                 }
 
                 return new InMemoryTableValue(node.IRContext, resultRows);
+            }
+            else if (node.Op == UnaryOpKind.RecordToRecord)
+            {
+                var fields = new List<NamedValue>();
+                var scopeContext = context.SymbolContext.WithScope(node.Scope);
+                var newScope = scopeContext.WithScopeValues((RecordValue)arg1);
+
+                foreach (var coercion in node.FieldCoercions)
+                {
+                    CheckCancel();
+
+                    var newValue = await coercion.Value.Accept(this, context.NewScope(newScope));

```

Comment:
Do we need a new scope? Can this just be context?

Reply:
Context doesn't contain all necessary record type scopes.